config/config.go

```diff
@@ -869,6 +881,21 @@ type PackageConfig struct {
 	Package string `json:"package"`
 	// Version of the package ID hosted by a remote PackageService. If not specified "latest" is assumed.
 	Version string `json:"version,omitempty"`
+	// Type is the type of package - for backward compatibility can be left empty
+	Type PackageType `json:"type,omitempty"`
+}
+
+// HashName forms the package name for the symlink/filepath of the package on the system.
+func HashName(f PackageConfig) string {
+	// replace / to avoid a directory path in the name. This will happen with `org/package` format.
+	// also makes the path valid by replacing all the dots with _
+	return fmt.Sprintf("%s-%s", strings.ReplaceAll(f.Package, "/", "-"), HashVersion(f.Version))
+}
+
+// HashVersion formats a version string to replace periods with dashes.
+func HashVersion(version string) string {
+	// replaces all the . if they exist with _
+	return strings.ReplaceAll(version, ".", "_")
```

Comment:
used to create the package path replacement so versioning is in a correct file format